tests/mgbench/benchmark.py

```diff
@@ -474,324 +477,340 @@ def get_query_cache_count(vendor, client, func, config_key):
+        sys.exit(0)
 
-    # TODO: Create some abstract class for vendors, that will hold this data
-    if args.vendor_name == "neo4j":
-        vendor = runners.Neo4j(
-            args.vendor_binary,
-            args.temporary_directory,
-            args.bolt_port,
-            args.performance_tracking,
-        )
+    # Create cache, config and results objects.
+    cache = helpers.Cache()
+    if not args.no_load_query_counts:
+        config = cache.load_config()
     else:
-        vendor = runners.Memgraph(
-            args.vendor_binary,
-            args.temporary_directory,
-            not args.no_properties_on_edges,
-            args.bolt_port,
-            args.performance_tracking,
-        )
+        config = helpers.RecursiveDict()
+    results = helpers.RecursiveDict()
+
+    # Filter out the generators.
+    benchmarks = filter_benchmarks(generators, args.benchmarks, args.vendor_name)
+    # Run all specified benchmarks.
+    for dataset, queries in benchmarks:
+
+        results.set_value("__run_configuration__", value=run_config)
 
-    client = runners.Client(args.client_binary, args.temporary_directory, args.bolt_port)
-
-    ret = None
-    usage = None
-    if args.vendor_name == "neo4j":
-        vendor.start_preparation("preparation")
-        print("Executing database cleanup and index setup...")
-        ret = client.execute(file_path=dataset.get_index(), num_workers=args.num_workers_for_import)
-        usage = vendor.stop("preparation")
-        dump_dir = cache.cache_directory("datasets", dataset.NAME, dataset.get_variant())
-        dump_file, exists = dump_dir.get_file("neo4j.dump")
-        if exists:
-            vendor.load_db_from_dump(path=dump_dir.get_path())
+        log.init("Preparing", dataset.NAME + "/" + dataset.get_variant(), "dataset")
+        dataset.prepare(cache.cache_directory("datasets", dataset.NAME, dataset.get_variant()))
+
+        # TODO: Create some abstract class for vendors, that will hold this data
+        if args.vendor_name == "neo4j":
+            vendor = runners.Neo4j(
+                args.vendor_binary,
+                args.temporary_directory,
+                args.bolt_port,
+                args.performance_tracking,
+            )
         else:
-            vendor.start_preparation("import")
-            print("Importing dataset...")
-            ret = client.execute(file_path=dataset.get_file(), num_workers=args.num_workers_for_import)
-            usage = vendor.stop("import")
+            vendor = runners.Memgraph(
+                args.vendor_binary,
+                args.temporary_directory,
+                not args.no_properties_on_edges,
+                args.bolt_port,
+                args.performance_tracking,
+            )
+
+        client = runners.Client(args.client_binary, args.temporary_directory, args.bolt_port)
+
+        ret = None
+        usage = None
+        ret, usage = importer.Importer(
```

Comment:
Let's think about this, about deps, draw dependencies, goals are:
* Make super easy to add new Workload
* Make it super easy to add custom Importer